firewood/src/storage/linear/mod.rs

```diff
@@ -96,3 +96,79 @@ impl<S: ReadLinearStore> ReadLinearStore for LinearStore<S> {
         todo!()
     }
 }
+
+#[allow(clippy::unwrap_used, clippy::indexing_slicing)]
+#[cfg(test)]
+pub mod tests {
+    use super::{ReadLinearStore, WriteLinearStore};
+    use std::io::Read;
+    use test_case::test_case;
+
+    #[derive(Debug)]
+    pub struct InMemReadWriteLinearStore {
+        bytes: Vec<u8>,
+    }
+
+    impl InMemReadWriteLinearStore {
+        pub const fn new() -> Self {
+            Self { bytes: vec![] }
+        }
+    }
+
+    impl WriteLinearStore for InMemReadWriteLinearStore {
+        fn write(&mut self, offset: u64, object: &[u8]) -> Result<usize, std::io::Error> {
+            let offset = offset as usize;
+            if offset + object.len() > self.bytes.len() {
+                self.bytes.resize(offset + object.len(), 0);
+            }
+            self.bytes[offset..offset + object.len()].copy_from_slice(object);
+            Ok(object.len())
+        }
+    }
+
+    impl ReadLinearStore for InMemReadWriteLinearStore {
+        fn stream_from(&self, addr: u64) -> Result<impl Read, std::io::Error> {
+            if addr as usize >= self.bytes.len() {
+                return Err(std::io::Error::new(
```

Comment:
Reads streaming past the eof should just return an eof, not an error. You can just return an empty cursor.

This happens when reading from a proposal that has extended the file size. It then switches back to the base image to look for bytes past that, which do not exist.

Reply:
done